### DISCUSSION TITLE
Global/User configuration

### DISCUSSION THREAD

**Discussion by kuchta** [biomejs/biome#2772] [General]
If would be handy to have some means of configuring biome on global/user level, at least if there isn't a project specific configuration...

**Comment by nhedger** [comment]
This is already how it works.

If Biome cannot find a `biome.json` file in the working directory, it will try to find one in the parent directories.

https://biomejs.dev/guides/how-biome-works/#configuration-file-resolution

**Reply by kuchta** [reply]
Thank you @nhedger . This is great and should solve most user's needs, but I think global configuration in $XDG_CONFIG_HOME or similar could be also great, as most users are probably used to that....

**Reply by OxiDoxi** [reply]
Biome can be used on individual files, which makes directory traversal not ideal

for example imagine I download a file and want to work on it in my download folder, am i expected to keep a `biome.json` in this folder?

the current way it works is great for organized projects, but not ideals for individual files on a device

**Reply by Araxeus** [reply]
That doesn't seem to work in vscode for some reason

**Reply by ematipico** [reply]
Biome accepts an environment variable called `BIOME_CONFIG_PATH`, which should cover it.

**Reply by Araxeus** [reply]
> Biome accepts an environment variable called `BIOME_CONFIG_PATH`, which should cover it.
> 

Doesn't seem like it does

Do you expect me to set this env variable globally? It would override the config for normal projects

Do you expect me to set this env manually on each single file i run that isn't in a project?

**Reply by ematipico** [reply]
@Araxeus 

Sorry, I wasn't answering directly to you, but to the original query.

I'm not sure what you're trying to do and what your expectations are, however the **VSCode extension** has an option that checks `$PATH`. Not sure if it helps.

**Reply by Araxeus** [reply]
> @ematipico 
> 
> Sorry, I wasn't answering directly to you, but to the original query.

What i said also applies to the original query:

If you set this env variable it will overwrite the local project config, while the assumption would be that the local project config would overwrite the global config

**Reply by ematipico** [reply]
But that's how it already works, as explained in the original reply. If that doesn't work in the VSCode extension, I suggest opening an issue with a reproduction

**Reply by Araxeus** [reply]
I've already opened an issue about the vscode problem last week

https://github.com/biomejs/biome-vscode/issues/298
